### PR TITLE
[KFS-1676] remove check for /dev/tty existence

### PR DESCRIPTION
**Pull request by YannickPferr**
Release Notes
-------------

Bug Fixes
- Fixes an issue where `confluent flink shell` didn't work on Windows
- Fixes an issue where pending statements couldn't be cancelled in `confluent flink shell`

Checklist
---------
- [x] Leave this box unchecked if features are not yet available in production

What
----

On Windows you will never be able to open /dev/tty, so this check doesn't really make sense here.

To still address the original intent of this check, I've updated go-prompt to return an error instead of panicking when it fails to access /dev/tty

**Note: we still need to officially change go-prompt and release a new tag, so I'm depending on a branch here. Will update once go-prompt was released**

References
----------

Test & Review
-------------

### CODE REVIEW

File: pkg/flink/internal/utils/input.go [hunk 1]
```diff
@@ -19,20 +19,17 @@ func GetStdin() *term.State {
...
-	consoleParser := prompt.NewStandardInputParser()
-	err := consoleParser.Setup()
+func GetConsoleParser() (prompt.ConsoleParser, error) {
+	consoleParser, err := prompt.NewStandardInputParser()
 	if err != nil {
+		log.CliLogger.Warnf("Couldn't create console parser: %v", err)
+		return nil, err
+	}
+	if err := consoleParser.Setup(); err != nil {
 		log.CliLogger.Warnf("Couldn't setup console parser: %v", err)
```
Comment (brianstrauch) [inline pkg/flink/internal/utils/input.go:29]:
```suggestion
		log.CliLogger.Warnf("Couldn't set up console parser: %v", err)
```

File: go.mod [hunk 2]
```diff
@@ -39,7 +39,7 @@ require (
 	github.com/confluentinc/confluent-kafka-go v1.9.3-RC3
 	github.com/confluentinc/go-editor v0.11.0
 	github.com/confluentinc/go-netrc v0.0.0-20220321173724-4d50f36ff450
-	github.com/confluentinc/go-prompt v0.2.26
+	github.com/confluentinc/go-prompt v0.2.27-0.20240213181507-604017e2bb08
```
Comment (YannickPferr) [inline go.mod:42]: will update this tomorrow once we've released a new go-prompt version

Comment (YannickPferr) [reply go.mod]: updated

**Review by Guttz — DISMISSED** [review]
nice

File: go.sum [hunk 3]
```diff
@@ -166,8 +166,12 @@ github.com/confluentinc/go-editor v0.11.0 h1:fcEALYHj7xV/fRSp54/IHi2DS4GlZMJWVgr
 github.com/confluentinc/go-editor v0.11.0/go.mod h1:nEjwqdqx8S7ZGjXsDvRgawsA04Fu2P/KAtA8fa5afMI=
 github.com/confluentinc/go-netrc v0.0.0-20220321173724-4d50f36ff450 h1:YJq9ZRhj049uo3hX7V8imJid+FJEBFyRmVcdruhU3y4=
 github.com/confluentinc/go-netrc v0.0.0-20220321173724-4d50f36ff450/go.mod h1:GGVB1yDj9YYQTxoo5vTRGWuQKccGN00bvyit4O5jznI=
-github.com/confluentinc/go-prompt v0.2.26 h1:DuzCAzU2dQoka8dJ0TiNDHXEbgIN5iVMZCLbSdtZgD4=
-github.com/confluentinc/go-prompt v0.2.26/go.mod h1:zevvJDLYjJBw/CFOd+9O1xBChio6z3WP7skZbw0YDC4=
+github.com/confluentinc/go-prompt v0.2.27-0.20240213181507-604017e2bb08 h1:Q9sQIhMeRxlNZaIaBBt21dyjzpNfIwaUxoYmVZVF828=
+github.com/confluentinc/go-prompt v0.2.27-0.20240213181507-604017e2bb08/go.mod h1:zevvJDLYjJBw/CFOd+9O1xBChio6z3WP7skZbw0YDC4=
+github.com/confluentinc/go-prompt v0.2.27 h1:OfodwY4NIYaF6aJ0W7S2vblijwZ7RJ+CkqEy5hDcoSY=
```
Comment (brianstrauch) [inline go.sum:171]: Wonder why 0.2.27 is still here. Run `go mod tidy`?

**Review by brianstrauch — DISMISSED** [review]

File: go.sum (hunk 3 above)
Comment (YannickPferr) [reply go.sum:171]: oh yeah forgot that

Comment (YannickPferr) [reply go.sum]: done

**Review by brianstrauch — APPROVED** [review]

**Pull request merged**